BaseCollector.py

```diff
@@ -0,0 +1,7 @@
+from abc import ABC, abstractmethod
+
+
+class BaseCollector(ABC):
+
+    def collect(self):
```

Comment:
I'd make it an `@abstractmethod`

Reply:
My mistake, imported abstractmethod, but forgot to use it.